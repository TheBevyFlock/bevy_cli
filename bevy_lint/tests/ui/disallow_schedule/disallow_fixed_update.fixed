#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::disallow_fixed_update)]

use bevy::prelude::*;

fn main() {
    let mut app = App::new();

    app.add_plugins(DefaultPlugins)
        //~| HELP: use the Update schedule instead
        //~v ERROR: use of the FixedUpdate schedule is disallowed
        .add_systems(Update, hello_world);

    // Ensure the lint can be muted by annotating the expression.
    #[allow(bevy::disallow_fixed_update)]
    app.add_systems(FixedUpdate, hello_world);

    app.run();
}

fn hello_world() {
    println!("hello world!");
}
