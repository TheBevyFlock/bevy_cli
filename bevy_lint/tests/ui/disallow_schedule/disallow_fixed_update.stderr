error: use of the `FixedUpdate` schedule is disallowed
  --> tests/ui/disallow_schedule/disallow_fixed_update.rs:13:22
   |
13 |         .add_systems(FixedUpdate, hello_world);
   |                      ^^^^^^^^^^^ help: use the `Update` schedule instead: `Update`
   |
note: the lint level is defined here
  --> tests/ui/disallow_schedule/disallow_fixed_update.rs:3:9
   |
3  | #![deny(bevy::disallow_fixed_update)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

